source/common/json/json_loader.cc

```diff
@@ -521,7 +523,7 @@ bool ObjectHandler::StartObject() {
     state_ = expectKeyOrEndObject;
     return true;
   default:
```

Comment:
not all states are in the case statement. the NOT_REACHED will only be hit if something unexpected happens in the state machine (e.g. due to unexpected sequence of calls from rapidjson).